### PR TITLE
Make `maker_cfd::Actor` generic over other actor addresses

### PR DESCRIPTION
**Pull request by luckysori**
The diff is gonna be hard to review because I had to reorganise all the methods in separate `impl` blocks so that we only require the necessary trait bounds across the board.

### CODE REVIEW

**Comment by thomaseizinger** [comment]
What is the reason for not resorting to `MessageChannel`s here as well? Would it be too verbose due to the number of messages?

**Comment by luckysori** [comment]
> What is the reason for not resorting to `MessageChannel`s here as well? Would it be too verbose due to the number of messages?

I think it would be a lot of channels, yep.

File: daemon/src/maker_cfd.rs [hunk 1]
```diff
@@ -58,19 +58,19 @@ pub struct FromTaker {
     pub msg: wire::TakerToMaker,
 }
 
-pub struct Actor {
+pub struct Actor<O, M, T> {
```
Comment (da-kami) [inline daemon/src/maker_cfd.rs:61]:
```suggestion
pub struct Actor<O, M, G> {
```

🤪

**Review by da-kami — COMMENTED** [review]
I actually find it somewhat nice to have the handling of specific messages grouped by the actors they relate to. But all that boilerplate code is just not very nice. @thomaseizinger any chance this can be optimized upstream, so we don't have to add all those `where` clauses?

File: daemon/src/maker_cfd.rs (hunk 1 above)
Comment (luckysori) [reply daemon/src/maker_cfd.rs:61]: image

**Comment by klochowicz** [comment]
> I actually find it somewhat nice to have the handling of specific messages grouped by the actors they relate to. But all that boilerplate code is just not very nice. @thomaseizinger any chance this can be optimized upstream, so we don't have to add all those `where` clauses?

we might give `spaad` crate another go (it's main goal is reducing this boilerplate...)

**Comment by da-kami** [comment]
> > I actually find it somewhat nice to have the handling of specific messages grouped by the actors they relate to. But all that boilerplate code is just not very nice. @thomaseizinger any chance this can be optimized upstream, so we don't have to add all those `where` clauses?
> 
> we might give `spaad` crate another go (it's main goal is reducing this boilerplate...)

that does look noice! https://crates.io/crates/spaad

**Review by luckysori — COMMENTED** [review]
LGTM. It's tricky to review, but I didn't notice anything wrong.

File: daemon/src/maker_cfd.rs [hunk 2]
```diff
@@ -492,7 +495,7 @@ impl Actor {
             }
         };
 
-        // 2. Create a new CFD
+        // 2. Insert CFD in DB
```
Comment (luckysori) [inline daemon/src/maker_cfd.rs:498]: :question: Is this comment supposed to change?

**Review by luckysori — COMMENTED** [review]
Did another pass with the two files side by side (the diff is impossible to follow in this case) and it still looks good. I think we can merge this.

File: daemon/src/maker_cfd.rs (hunk 2 above)
Comment (thomaseizinger) [reply daemon/src/maker_cfd.rs:498]: Uhm I just made the same semantic change I think and just rewrote the comment.

**Comment by thomaseizinger** [comment]
> Did another pass with the two files side by side (the diff is impossible to follow in this case) and it still looks good. I think we can merge this.

Go for it!

**Pull request merged**